src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -27,21 +27,21 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             StorageCredentials credentials = account.Credentials;
 
             // Avoid double-validating the same account and credentials.
-            if (_validatedCredentials.Contains(credentials))
+            if ((!isPrimaryAccount && _validatedCredentials.Contains(credentials)) || _primaryCredentials == credentials)
             {
                 return;
             }
 
-            await ValidateCredentialsAsyncCore(account, cancellationToken);
+            await ValidateCredentialsAsyncCore(account, cancellationToken, isPrimaryAccount);
             _validatedCredentials.Add(credentials);
         }
 
-        private static async Task ValidateCredentialsAsyncCore(IStorageAccount account,
-            CancellationToken cancellationToken)
+        private async Task ValidateCredentialsAsyncCore(IStorageAccount account,
+            CancellationToken cancellationToken, bool isPrimaryAccount)
         {
             // Verify the credentials are correct.
             // Have to actually ping a storage operation.
-            IStorageBlobClient client = account.CreateBlobClient();
+            var client = account.CreateBlobClient();
```

Comment:
Revert this change. As a rule of thumb, if the type is not clear (as in the result of a method call), you want to be explicit about the type.